### DISCUSSION TITLE
Can't "prefer_gpu()"

### DISCUSSION THREAD

**Discussion by Mathhead** [explosion/spaCy#10441] [Help: Installation] [answered]
Hi there.

I am trying to run spaCy on my GPU, yet "prefer_gpu()" returns False.

## How to reproduce the behaviour
!pip install -U 'spacy[cuda114]'
import spacy
spacy.prefer_gpu()

## Your Environment
2022-03-03 15:48:07.197671: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2022-03-03 15:48:07.201202: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.6

## Info about spaCy

- **spaCy version:** 3.2.3
- **Platform:** Linux-5.4.0-99-generic-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.6.9
-**CUDA 11.4
-**Driver Version: 470.63.01 
-**tf.config.list_physical_devices(): [PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'),
 PhysicalDevice(name='/physical_device:XLA_CPU:0', device_type='XLA_CPU'),
 PhysicalDevice(name='/physical_device:XLA_GPU:0', device_type='XLA_GPU'),
 PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]

**Comment by Mathhead** [comment]
Hi persons.


Could you please give me any advice on what information I could deliver to help you help me?

I am trying to run spacy on a virtual GPU. Does this help you?

**Comment by polm** [comment, marked as answer]
Check these imports:

```
import cupy
import cupyx
import cupy.cuda
from cupy.cuda.compiler import compile_with_cache
import torch
```

If any of those fail, that could cause this. 

Also I note you are using Python 3.6. I think that should still work in theory, but it did reach end-of-life last year; I recommend upgrading to more recent version.

**Reply by Mathhead** [reply]
Thank you.

It seems that I have problems with cupy. 

I cannot uninstall it properly, so I think, there is something horribly wrong with my installation. Therefore, I figure, I should sort out my problem with cupy first.
